File: scripts/theme.py
"""Single source of truth for transit line and administrative layer styling.

Reads the same JSON definitions used by the web app (web/src/*.json) so the
interactive map and the generated PDFs stay in sync.
"""

from __future__ import annotations

import json
from pathlib import Path

_SRC_DIR = Path(__file__).resolve().parents[1] / "web" / "src"

TRANSIT_LINES: list[dict] = json.loads((_SRC_DIR / "transit-lines.json").read_text())
ADMIN_LAYERS: list[dict] = json.loads((_SRC_DIR / "admin-layers.json").read_text())

LINE_COLORS: dict[str, str] = {line["line"]: line["color"] for line in TRANSIT_LINES}


def lines_for_network(network: str) -> list[dict]:
    return [line for line in TRANSIT_LINES if line["network"] == network]

File: web/src/theme.js
import transitLines from './transit-lines.json'
import adminLayers from './admin-layers.json'

// Single source of truth for transit line styling and metadata.
// Consumed by the interactive map, the guide page, and (via web/src/*.json)
// the Python PDF generators.
export const TRANSIT_LINES = transitLines
export const ADMIN_LAYERS = adminLayers

export const LINE_COLORS = Object.fromEntries(
  transitLines.map((line) => [line.line, line.color]),
)

export function linesForNetwork(network) {
  return transitLines.filter((line) => line.network === network)
}

export function adminLayerStyle(layer) {
  return {
    color: layer.color,
    fillColor: layer.color,
    weight: 2.8,
    opacity: 0.8,
    fillOpacity: 0.12,
  }
}
